docs/types/mapping-types.rst

```diff
@@ -53,7 +53,100 @@ each ``_KeyType``, recursively. For example with a mapping:
         }
     }
 
+.. index:: !iterable mappings
+.. _iterable-mappings:
 
-.. note::
-  Mappings are not iterable, but it is possible to implement a data structure
-  on top of them. For an example, see `iterable mapping <https://github.com/ethereum/dapp-bin/blob/master/library/iterable_mapping.sol>`_.
+Iterable Mappings
+-----------------
+
+Mappings are not iterable, but it is possible to implement a data structure on
+top of them and iterate over that. For example, the code below implements an
+``IterableMapping`` library that the ``User`` contract then adds data too, and
```

Comment:
```suggestion
``IterableMapping`` library that the ``User`` contract then adds data to, and
```
?

Reply:
Nope, "too" is kind of like "in addition"

Reply:
Ah ok!